query-engine/core/src/query_graph_builder/read/utils.rs

```diff
@@ -258,6 +259,7 @@ pub(crate) fn get_relation_load_strategy(
             ReadQuery::RelatedRecordsQuery(q) => q.has_cursor() || q.has_distinct() || q.has_aggregation_selections(),
             _ => false,
         })
+        && requested_strategy != Some(RelationLoadStrategy::Query)
```

Comment:
Nit: (maybe that does not change much), it would be "faster" to check for this condition earlier (maybe in second position?).
The other conditions would not be evaluated then when  RelationLoadStrategy is set to `query`

Reply:
Hard to tell if it matters without looking at the generated code but yeah it makes sense even from just from stylistic perspective, I think the code reads a bit more naturally with this check earlier. We need to get this merged now as we don't have much time waiting on CI before the release but I can change it in a follow up PR.